Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java

```diff
@@ -150,7 +160,7 @@ public void onResponse(String deviceData) {
             @Override
             public void onResponse(Boolean isReadyToPay) {
                 mSupportedPaymentMethodListView.setAdapter(new SupportedPaymentMethodsAdapter(
-                        DropInActivity.this, configuration, isReadyToPay, mClientTokenPresent,
+                        DropInActivity.this, configuration, isReadyToPay && mDropInRequest.isAndroidPayEnabled(), mClientTokenPresent,
```

Comment:
If we're disabling Android Pay we shouldn't spend any time in `AndroidPay`

I think we can just wrap the whole `AndroidPay.isReadyToPay(...)` method in an if block checking if Android Pay is enabled:

```
if (mDropInRequest.isAndroidPayEnabled()) {
  AndroidPay.isReadyToPay(...);
}
```